src/DotLiquid/Tags/Capture.cs

```diff
@@ -22,7 +22,7 @@ namespace DotLiquid.Tags
     /// </summary>
     public class Capture : DotLiquid.Block
     {
-        private static readonly Regex Syntax = R.C(@"(\w+)");
+        private static readonly Regex Syntax = R.B(R.Q(@"({0}+)"), Liquid.VariableSignature);

```

Comment:
Variable signature includes dotted and bracketed notations. This should be a simple variable name imho. Other regex just include hyphen.